### ISSUE TITLE
faker.internet.url() returning invalid URLs with spaces

### ISSUE THREAD

**Issue by cduff** [faker-js/faker#256]
### Describe the bug

faker.internet.url() sometimes returns invalid URLs with spaces, e.g.

http://sick-dune buggy.name
https://hollow-dust storm.info

This is causing some of my tests which incorporate URL validation to intermittently fail. I had never seen this with the old "faker" but it started happening since moving to "@faker-js/faker" (6.0.0-alpha.3).

### Reproduction

``` TypeScript
for (let i = 0; i < 1000; i++) {
  const url = faker.internet.url();
  if (url.includes(" ")) console.log(url);
  // Yields unexpected output of invalid URLs
}
```

### Additional Info

https://www.npmjs.com/package/@faker-js/faker/v/6.0.0-alpha.3

**Comment by ST-DDT** [comment]
I can confirm that it fails.
I'll have a look at it.

**Comment by ST-DDT** [comment]
I found a fix for this, but AFAICT the bug should have been present before faker-js took this over:

- https://github.com/faker-js/faker/blob/29234378807c4141588861f69421bf20b5ac635e/lib/internet.js#L155
- https://github.com/faker-js/faker/blob/29234378807c4141588861f69421bf20b5ac635e/lib/internet.js#L172
- https://github.com/faker-js/faker/blob/29234378807c4141588861f69421bf20b5ac635e/lib/internet.js#L200
- https://github.com/faker-js/faker/blob/29234378807c4141588861f69421bf20b5ac635e/lib/locales/en/word/noun.js#L1967

It was probably introduced here:
https://github.com/faker-js/faker/commit/3aac69e8ab0d0337580de688a8e0134a20fdcc08#diff-970ae648187190f86bafc8f193b7538200eba164fad0674428b6487582c089ccR200

**Comment by Shinigami92** [comment]
Seems it was introduced in `6.0.0-alpha.0` :thinking: 
But it's a "new" feature :thinking: 
Maybe we should rollback and introduce the change later in v6.1?

**Comment by ST-DDT** [comment]
> Maybe we should rollback and introduce the change later in v6.1?

We don't know how many hidden and untested changes/bugs are there, so we shouldn't revert the code to introduce it again in the next release. If there are errors, we should just fix them, when they are reported.

**Comment by Shinigami92** [comment]
Okay, but then I would vote to provide this bug at least in v6.0.1
I currently would very like to not mess around with the runtime code until we released v6.0.0

**Comment by schw4rzlicht** [comment]
This is a huge issue for my team too b/c currently we have to decide between `@types/faker` and `@faker-js/faker@next` - we decided for the latter and now our tests fail regularly 😬

But having it in 6.0 means that it's considered urgent, right? How could I possibly help with fixing that issue?

**Comment by ST-DDT** [comment]
There is already #258 to fix it.

**Comment by Shinigami92** [comment]
Yeah, we just need only some one more review and then we could ask @damienwebdev to make a `6.0.0-alpha.6` release
Like the avatar URL fix, this counts to the critical bugs I think

**Comment by Shinigami92** [comment]
@schw4rzlicht as a workaround, could you try to call just `faker.internet.domainWord().replace(/\s/g, '-').replace(/-{2,}/g, '-')` :thinking:

**Comment by schw4rzlicht** [comment]
> @schw4rzlicht as a workaround, could you try to call just `faker.internet.domainWord().replace(/\s/g, '-').replace(/-{2,}/g, '-')` 🤔

Thanks for the proposal, but we have ~500 tests that may or may not depend on it 😄 I'd rather wait for `6.0.0-alpha.6` or migrate back to <6.

@damienwebdev any news on the next alpha release?

**Comment by schw4rzlicht** [comment]
What we just ended up with is basically replacing `faker.internet.domainName()` globally for the time being, thanks @Shinigami92 for the inspiration:

```typescript
// @ts-ignore
faker.internet.legacyDomainName = faker.internet.domainName;

// @ts-ignore
faker.internet.domainName = () => faker.internet.legacyDomainName().replace(/\s/g, "-").replace(/-{2,}/g, "-");
```

**Comment by Shinigami92** [comment]
@schw4rzlicht Did you know that we released 6.0.0-alpha.6 some hours ago? :smiley:

**Comment by schw4rzlicht** [comment]
No I didn't 😱 Just upgraded, works like a charm. Thank you all for your great work!
